### PR TITLE
Skip GitLab MR reply comments

### PR DESCRIPTION
**Pull request by Half-Shot**
GitLab uses `discussion_id` to chain together replies. We can't really tell what position we are *in* the thread, so we just have to assume that if we've not seen the discussion before that we are the first note, and subsequent notes are child notes.

This also changes the code so that we skip sending a notice into the room if the user has disabled comment-only notices.

### CODE REVIEW

File: src/Connections/GitlabRepo.ts [hunk 1]
```diff
@@ -285,8 +286,16 @@ export class GitLabRepoConnection extends CommandConnection<GitLabRepoConnection
         author: string,
         timeout: NodeJS.Timeout,
         approved?: boolean,
+        skip?: boolean,
     }>();
 
+    /**
+     * GitLab provides NO threading information in it's webhook response objects,
```
Comment (jaller94) [inline src/Connections/GitlabRepo.ts:293]:
```suggestion
     * GitLab provides NO threading information in its webhook response objects,
```

File: src/Connections/GitlabRepo.ts [hunk 2]
```diff
@@ -671,23 +688,42 @@ ${data.description}`;
...
+                skip: false,
             }
         );
     }
 
+    private shouldHandleMRComment(event: IGitLabWebhookNoteEvent) {
+        // Check to see if this line has had a comment before
+        if (event.object_attributes.discussion_id) {
+            if (this.mergeRequestSeenDiscussionIds.has(event.object_attributes.discussion_id)) {
+                // If it has, this is probably a reply. Replies are noise, skip em.
```
Comment (jaller94) [inline src/Connections/GitlabRepo.ts:701]:
```suggestion
                // If it has, this is probably a reply. Skip repeated replies.
```

File: src/Connections/GitlabRepo.ts [hunk 3]
```diff
@@ -642,13 +655,17 @@ ${data.description}`;
             if (commentNotes) {
                 existing.commentNotes = [...(existing.commentNotes ?? []), ...commentNotes];
             }
-            existing.commentCount = opts.commentCount;
+            existing.commentCount = existing.commentCount + opts.commentCount;
```
Comment (jaller94) [inline src/Connections/GitlabRepo.ts:658]: This suggestion is just fancy syntax sugar. No need to apply.
```suggestion
            existing.commentCount += opts.commentCount;
```

**Review by jaller94 — APPROVED** [review]
Looks good to me.

I'm a bit surprised that Hookshot depends on `@alloc/quick-lru` instead of the upstream `quick-lru` (which became an ESM in version 6).

**Comment by Half-Shot** [comment]
>I'm a bit surprised that Hookshot depends on @alloc/quick-lru instead of the upstream quick-lru (which became an ESM in version 6).

We can't use ESM, as it introduces a bunch of annoying incompatibilities. Instead of trying to refactor the entire project to be ESM compatible, we've settled for just using a cJS library for the moment.

**Pull request merged**